js/adapt-hint.js

```diff
@@ -1,120 +1,102 @@
-				}, {
-					duration: 300,
-					display: 'none'
-				});
-				$specDetail.removeClass('widget-open');
-
-				this.$('.hint-extension-button').removeClass('icon-cross').addClass('icon-question');
-				 Adapt.trigger('popup:closed',  this.$('.hint-extension-inner'));
-			}
-		},
-
-		checkIfShouldClose: function(id) {
-			if (this.model.get('_id') !== id) {
-				var $widget = $('.' + this.model.get('_id') + " .hint-extension-widget");
-				var $button = $('.' + this.model.get('_id') + " .hint-extension-button");
-
-				$widget.velocity({
-					scaleX: 0,
-					scaleY: 0
-				}, {
-					duration: 300,
-					display: 'none'
-				});
-
-				$widget.removeClass('widget-open');
-				$button.removeClass('icon-cross').addClass('icon-question');
-			}
-		}
-	});
-
-	Adapt.on('componentView:postRender', function(view) {
-		if (view.model.has('_hint') && view.model.get('_hint').length > 0) {
-			new hintExtensionView({
-				model: view.model
-			});
-		}
-	});
+define([
+  'core/js/adapt'
+], function (Adapt) {
+
+  var HintPopupView = Backbone.View.extend({
+    className: 'hint-popup',
+
+    initialize: function() {
+      this.render();
+    },
+
+    render: function() {
+      const data = this.model.toJSON();
+      data.view = this;
+      const template = Handlebars.templates.hintPopup;
+      this.$el.html(template(data));
+    }
+  });
+
+  var HintView = Backbone.View.extend({
+
+    events: {
+      'click .js-hint-btn-toggle': 'onHintToggleClicked',
+      'click .js-hint-btn-popup': 'onHintPopupClicked'
+    },
+
+    className: 'hint',
+
+    initialize: function() {
+      this.listenTo(Adapt, 'remove', this.remove);
+
+      if (!this.model.get('_hint')._isNotifyPopup) {
+        this.listenTo(Adapt, 'hint:opened', this.checkIfShouldClose);
+      }
+
+      this.render();
+    },
+
+    render: function() {
+      const data = this.model.toJSON();
+      const template = Handlebars.templates.hint;
+
+      this.$el.html(template(data)).appendTo($('.' + this.model.get('_id')));
+
+      _.defer(this.postRender.bind(this));
+    },
+
+    postRender: function() {
+      this.$el.parents('.component').addClass('has-hint');
+    },
+
+    onHintPopupClicked: function() {
+      Adapt.notify.popup({
+        _view: new HintPopupView({ model: this.model }),
+        _isCancellable: true,
+        _showCloseButton: true,
+        _closeOnBackdrop: true
+      });
+    },
+
+    onHintToggleClicked: function (e) {
+      const $button = this.$('.js-hint-btn-toggle');
+      const $widget = this.$('.js-hint-widget');
+      const globals = Adapt.course.get('_globals')._extensions._hint;
+      const isBeingOpened = $widget.hasClass('is-closed');
+
+      $widget.toggleClass('is-open', isBeingOpened).toggleClass('is-closed', !isBeingOpened);
+      $button.attr('aria-label', (isBeingOpened ? globals.closeButtonText : globals.openButtonText))
+        .toggleClass('is-open', isBeingOpened).toggleClass('is-closed', !isBeingOpened);
```

Comment:
As the modification classes are now identical, can the Less be reworked to use them from a shared parent higher up in the DOM, @guywillis?

For example, with the below or similar:
```js
this.$el
  .toggleClass('is-open', isBeingOpened)
  .toggleClass('is-closed', !isBeingOpened);
```

Reply:
Whilst it would make for more optimised code, making this change would make the HTML and Less slightly less easier to read in my opinion. 

I've left the current implementation but made the code a little easier to read.